drcp-format/src/reader/mod.rs

```diff
@@ -306,3 +303,24 @@ fn identifier(input: &str) -> IResult<&str, &str> {
         many0_count(alt((alphanumeric1, tag("_")))),
     ))(input)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn inference_nogood_with_constraint_tag_and_label() {
```

Comment:
Would be good to add 1 test case where nothing is added but the `<step_id>` and the `<premises>`